engine/database/models.cc

```diff
@@ -0,0 +1,309 @@
+#include "models.h"
+#include <algorithm>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <stdexcept>
+#include "database.h"
+#include "utils/file_manager_utils.h"
+#include "utils/result.hpp"
+#include "utils/scope_exit.h"
+
+namespace cortex::db {
+
+Models::Models() : db_(cortex::db::Database::GetInstance().db()) {
+  db_.exec(
+      "CREATE TABLE IF NOT EXISTS models ("
+      "model_id TEXT PRIMARY KEY,"
+      "author_repo_id TEXT,"
+      "branch_name TEXT,"
+      "path_to_model_yaml TEXT,"
+      "model_alias TEXT,"
+      "status TEXT);");
+}
+
+Models::Models(SQLite::Database& db) : db_(db) {
+  db_.exec(
+      "CREATE TABLE IF NOT EXISTS models ("
+      "model_id TEXT PRIMARY KEY,"
+      "author_repo_id TEXT,"
+      "branch_name TEXT,"
+      "path_to_model_yaml TEXT,"
+      "model_alias TEXT UNIQUE,"
+      "status TEXT);");
```

Comment:
maybe we should remove status here
If we have the record inside db, we know that it's ready. other model status (loaded or something) will be stored in memory.